diesel/src/sqlite/types/date_and_time/chrono.rs

```diff
@@ -361,4 +434,21 @@ mod tests {
         let query = select(datetime("9999-01-08 00:00:00.000000").eq(distant_future));
         assert!(query.get_result::<bool>(connection).unwrap());
     }
+
+    #[test]
```

Comment:
These aren't passing now but I think I can figure that out when I come back to this 
```
@ diesel [stevenchu]$ cargo test --no-default-features --features "extras sqlite"
    Finished test [unoptimized + debuginfo] target(s) in 0.11s
     Running unittests (/Users/stevenchu/personal_projects/rust/diesel/target/debug/deps/diesel-8fccd0801605fa3c)

running 40 tests
test query_builder::query_id::tests::bind_params_use_only_sql_type_for_query_id ... ok
test query_builder::query_id::tests::queries_with_different_types_have_different_ids ... ok
test query_builder::query_id::tests::queries_with_no_dynamic_elements_have_a_static_id ... ok
test connection::transaction_manager::test::transaction_manager_enters_broken_state_when_connection_is_broken ... ok
test connection::commit_error_processor::tests::check_default_process_commit_error_implementation ... ok
test query_builder::select_statement::boxed::tests::boxed_is_send ... ok
test macros::tests::table_renaming_sqlite ... ok
test macros::tests::table_with_column_renaming_sqlite ... ok
test connection::transaction_manager::test::sqlite_transaction_is_rolled_back_upon_syntax_error ... ok
test sqlite::connection::tests::prepared_statements_are_cached_when_run ... ok
test connection::commit_error_processor::tests::check_invalid_transaction_state_rejected - should panic ... ok
test sqlite::connection::tests::queries_containing_in_with_subselect_are_cached ... ok
test r2d2::tests::is_valid ... ok
test connection::transaction_manager::test::sqlite_transaction_is_rolled_back_upon_deferred_constraint_failure ... ok
test r2d2::tests::pooled_connection_impls_connection ... ok
test r2d2::tests::establish_basic_connection ... ok
test sqlite::connection::tests::queries_containing_in_with_vec_are_not_cached ... ok
test sqlite::connection::row::fun_with_row_iters ... ok
test sqlite::connection::tests::register_custom_function ... ok
test sqlite::connection::tests::register_aggregate_multiarg_function ... ok
test sqlite::connection::tests::register_aggregate_function ... ok
test sqlite::connection::tests::queries_containing_sql_literal_nodes_are_not_cached ... ok
test sqlite::connection::tests::register_noarg_function ... ok
test sqlite::connection::tests::register_multiarg_function ... ok
test sqlite::connection::tests::register_aggregate_function_returns_finalize_default_on_empty_set ... ok
test sqlite::connection::tests::register_nondeterministic_function ... ok
test sqlite::connection::tests::register_collation_function ... ok
test sqlite::connection::tests::register_nondeterministic_noarg_function ... ok
test sqlite::connection::tests::sql_literal_nodes_are_not_cached ... ok
test sqlite::types::date_and_time::chrono::tests::dates_decode_correctly ... ok
test sqlite::types::date_and_time::chrono::tests::dates_encode_correctly ... ok
test sqlite::types::date_and_time::chrono::tests::datetimes_decode_correctly ... ok
test sqlite::types::date_and_time::chrono::tests::times_of_day_encode_correctly ... ok
test sqlite::types::date_and_time::chrono::tests::times_relative_to_now_encode_correctly ... ok
test sqlite::types::date_and_time::chrono::tests::unix_epoch_encodes_correctly ... ok
test sqlite::types::date_and_time::chrono::tests::unix_epoch_encodes_correctly_with_timezone ... FAILED
test sqlite::types::date_and_time::chrono::tests::datetimes_encode_correctly ... ok
test sqlite::types::date_and_time::chrono::tests::unix_epoch_decodes_correctly_with_timezone ... FAILED
test sqlite::types::date_and_time::chrono::tests::times_of_day_decode_correctly ... ok
test sqlite::types::date_and_time::chrono::tests::unix_epoch_decodes_correctly_in_all_possible_formats ... ok

failures:

---- sqlite::types::date_and_time::chrono::tests::unix_epoch_encodes_correctly_with_timezone stdout ----
thread 'sqlite::types::date_and_time::chrono::tests::unix_epoch_encodes_correctly_with_timezone' panicked at 'a Display implementation returned an error unexpectedly: Error', /rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/alloc/src/string.rs:2383:14

---- sqlite::types::date_and_time::chrono::tests::unix_epoch_decodes_correctly_with_timezone stdout ----
thread 'sqlite::types::date_and_time::chrono::tests::unix_epoch_decodes_correctly_with_timezone' panicked at 'assertion failed: `(left == right)`
  left: `Ok(1970-01-01T00:00:00Z)`,
 right: `Err(DeserializationError("Invalid datetime 1970-01-01Z"))`', diesel/src/sqlite/types/date_and_time/chrono.rs:452:9


failures:
    sqlite::types::date_and_time::chrono::tests::unix_epoch_decodes_correctly_with_timezone
    sqlite::types::date_and_time::chrono::tests::unix_epoch_encodes_correctly_with_timezone

test result: FAILED. 38 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s

error: test failed, to rerun pass '--lib'
```

Reply:
😅  I think I figured this one out with the good help of `println!()` debugging; I had to change the signature of the `ToSql` implementation to

```
impl<TZ: TimeZone> ToSql<TimestamptzSqlite, Sqlite> for DateTime<TZ> {
    fn to_sql<'b>(&'b self, out: &mut Output<'b, '_, Sqlite>) -> serialize::Result {
        out.set_value(self.naive_utc().format("%F %T+00:00").to_string());
        Ok(IsNull::No)
    }
}
```

resolving